src/components/decorative-boxes/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { useRef } from 'react';
+import style from './style.module.css';
+type BoxProperties = {
+  color: `--color-${DefaultColors}${ColorModifiers}` | HexColorAsString;
+  position: `${VerticalPosition}-${HorizontalPosition}` | NumericPosition;
+  text?: string | undefined;
+};
+
+type NumericPosition = {
+  horizontalPosition: number;
+  verticalPosition: number;
+};
+type BoxSize = 'small' | 'medium' | 'large' | number;
+type VerticalPosition = 'top' | 'topish' | 'middle' | 'bottomish' | 'bottom';
+type HorizontalPosition = 'left' | 'leftish' | 'middle' | 'rightish' | 'right';
+type DefaultColors = 'primary' | `secondary${1 | 2 | 3 | 4}`;
+type ColorModifiers = '' | `__${'tint' | 'shade'}${1 | 2 | 3 | 4}`;
+type HexColorAsString = `#${string}`;
+
+export interface DecorativeBoxesProps
+  extends React.ComponentPropsWithoutRef<any> {
+  children: JSX.Element;
+  boxSize?: BoxSize;
+  box1Properties: BoxProperties;
+  box2Properties: BoxProperties;
+}
+
+const DecorativeBoxes: React.FC<DecorativeBoxesProps> = ({
+  children,
+  boxSize = 'medium',
+  box1Properties,
+  box2Properties,
+}) => {
+  const box1HorizontalDistance = useRef(
```

Comment:
Jeg ville ikke ha brukt `useRef` her. Ref er primært et verktøyf for å oppnå "mutabilitet" på tvers av renders, altså der du ikke kan ha nye verdier av ting men må ha den gamle. Skjønner tanken om å gjøre performance her, men det er så lite som kjører at faktisk mistenker jeg at useRef vil gjøre mer skade enn nytte. Det vanlige dersom man ikke vil den skal kalkulere på nytt er bruk av `useMemo`, men selv det mener jeg er unødvendig her. Generelt så er "useMemo" noe som burde innføres når det er et reelt problem.

Reply:
Okay, så er alternativet bare å bruke en helt vanlig variabel? Altså å gå fra

`const box1HorizontalDistance = useref( foo() )`
til
`const box2HorizontalDistance = foo()`

Er en del av meg om er redd for at foo() vil kjøres maaange ganger flere enn nødvendig, men har egentlig ikke god nok kontroll på akkurat hvilken kode React kjører "hele tida" og ikke.

Reply:
Stemmer. Det skal re-rendres mye før det blir et problem her. Husk at det kommer fra serveren og det er ikke mye oppdatering av treet her, så det kjører nok ikke hyppig i det hele tatt.

Reply:
Da blir det bare gode gamle consts, uten noe jålete ekstrakode 😤